pkg/ibm_plugin/sdk/vpn.go

```diff
@@ -0,0 +1,596 @@
+	// delete all routes with routing next hop pointing at the specified connection
+	for _, route := range routeCollection.Routes {
+		routeNextHop, isNextHopToVpnConnection := route.NextHop.(*vpcv1.RouteNextHop)
+		if !isNextHopToVpnConnection {
+			return fmt.Errorf("Expected next hop to reference a VPN connection, instead (likely) references an IP address.")
+		}
+		if *routeNextHop.ID == *connection.ID {
+			_, err = c.vpcService.DeleteVPCRoute(&vpcv1.DeleteVPCRouteOptions{
+				VPCID: &vpcID,
+				ID:    route.ID,
+			})
+			// keep track of routes set for deletion (directing to the specified connection)
+			deletedRoutes = append(deletedRoutes, route)
+			if err != nil {
+				return err
+			}
+			utils.Log.Printf("Deleted VPC route %v directing traffic from %v ", *route.ID, *route.Zone.Name)
+		}
+	}
+
+	// wait for routes to delete
+	for _, route := range deletedRoutes {
+		err := c.pollRouteDeleted(vpcID, *defaultRoutingTable.ID, route)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// deletes the specified VPN along with its connections their associated routes
+func (c *CloudClient) DeleteVPN(VPNGatewayID string) error {
+	vpnConnections, _, err := c.vpcService.ListVPNGatewayConnections(
+		&vpcv1.ListVPNGatewayConnectionsOptions{VPNGatewayID: core.StringPtr(VPNGatewayID)})
+	if err != nil {
+		return err
+	}
+
+	// invoke delete operation on all VPN connections
+	for _, connectionInterface := range vpnConnections.Connections {
+		connection := connectionInterface.(*vpcv1.VPNGatewayConnection)
+		// delete routes directing to this connection
+		err := c.DeleteRoutesDependentOnConnection(VPNGatewayID, connection)
+		if err != nil {
+			return err
+		}
+		// set connection for deletion
+		_, err = c.vpcService.DeleteVPNGatewayConnection(
+			&vpcv1.DeleteVPNGatewayConnectionOptions{VPNGatewayID: &VPNGatewayID, ID: connection.ID})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	// wait for connections deletion operation to finalize
+	for _, connectionInterface := range vpnConnections.Connections {
+		connectionID := *connectionInterface.(*vpcv1.VPNGatewayConnection).ID
+		err = c.pollVPNConnectionDeleted(VPNGatewayID, connectionID)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = c.vpcService.DeleteVPNGateway(&vpcv1.DeleteVPNGatewayOptions{ID: &VPNGatewayID})
+	if err != nil {
+		return err
+	}
+	utils.Log.Printf("VPN gateway with ID %v was set for deletion", VPNGatewayID)
+
+	// wait for VPN deletion (can't delete reliant resources such as subnets otherwise)
+	err = c.pollVPNStatus(VPNGatewayID, false)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// return ResourceData object of a VPN gateway matching the specified namespace and region.
+// if region value is empty avoid filtering results by region.
+func (c *CloudClient) GetVPNsInNamespaceRegion(namespace, region string) ([]ResourceData, error) {
+	queryFilter := ResourceQuery{}
+	if len(region) > 0 {
+		queryFilter.Region = region
+	}
```

Comment:
```suggestion
	queryFilter.Region = region
```
I don't believe the length check is necessary